src/main/java/com/google/sps/data/DBInterface.java

```diff
@@ -0,0 +1,227 @@
+   */
+  public void deleteUser(String userId);
+
+  /**
+   * Sets a property's value to "true" for a certain user document.
+   * Can be used to let a user add a recipe to saved or created, or to let user follow a tag.
+   * @param userId the user's Firebase ID
+   * @param objectId the ID of either a recipe if the intent is to save/create, or of a tag for tag
+   *     following.
+   * @param collection a KEY constant from User class indicating which mode -- save, create, or
+   *     follow tag.
+   */
+  public void makeUserPropertyTrue(String userId, String objectId, String collection);
+
+  /**
+   * Deletes a property value for a certain user document.
+   * Can be used to let a user delete a recipe from saved or created, or to let user unfollow a tag.
+   * @param userId the user's Firebase ID
+   * @param objectId the ID of either a recipe if the intent is to unsave/create, or of a tag for
+   *     tag unfollowing.
+   * @param collection a KEY constant from User class indicating which mode -- save, create, or tag.
+   */
+  public void deleteUserProperty(String userId, String objectId, String collection);
+
+  /**
+   * Returns a list of all recipe metadata with any of the tags in the tag IDs list.
+   * @param tagIds tags' Firestore IDs
+   * @param sortingMethod such as TOP or NEW
+   * @return list of recipe metadata matching one or more of the tags in tagIds param
+   */
+  public List<RecipeMetadata> getRecipesMatchingTags(
+      List<String> tagIds, SortingMethod sortingMethod);
+
+  /**
+   * Returns a list of the recipe metadata with a certain creator.
+   * @param creatorId creator/user's Firestore ID
+   * @param sortingMethod such as TOP or NEW
+   * @return user's created recipe metadata, sorted
+   */
+  public List<RecipeMetadata> getRecipesMatchingCreator(
+      String creatorId, SortingMethod sortingMethod);
+
+  /**
+   * Returns a list of the recipe metadata associated with the IDs of recipes saved by a user.
+   * @param userId user's Firestore ID
+   * @param sortingMethod such as TOP or NEW
+   * @return user's saved recipe metadata, sorted
+   */
+  public List<RecipeMetadata> getRecipesSavedBy(String userId, SortingMethod sortingMethod);
+
+  /**
+   * Returns a list of the recipe metadata whose IDs match any one ID in a given list of IDs.
+   * @param Ids list of recipe Firestore IDs to be queried
+   * @param sortingMethod such as TOP or NEW
+   * @return list of RecipeMetadata whose IDs are in Ids param
+   */
+  public List<RecipeMetadata> getRecipesMatchingIDs(List<String> Ids, SortingMethod sortingMethod);
+
+  /**
+   * Returns an ordered list of RecipeMetadata objects based on a sorting method.
+   * @param recipesQuery recipe metadata collection query
+   * @param sortingMethod sorting method from SortingMethod class
+   * @return ordered list of recipe metadata
+   */
+   public List<RecipeMetadata> getRecipeMetadataQuery(Query recipesQuery, SortingMethod sortingMethod);
```

Comment:
We shouldn't be exposing any of the Firestore API to the outside. Frankly, this should probably be kept private in the FirestoreDB class, and not be in here at all.